src/components/image/index.tsx

```diff
@@ -52,6 +53,20 @@ function getContentApiDimensions(
 	return null
 }
 
+function getTheme(src: string): GlobalThemeOption | undefined {
+	let theme
+	const url = new URL(src, 'https://developer.hashicorp.com')
+	const themedImageSuffix = new RegExp(/#hide-on-(dark|light)/)
+
+	if (themedImageSuffix.test(url.hash)) {
+		const match = url.hash.match(themedImageSuffix)
```

Comment:
Any thoughts for/against throwing in some sort of guardrails (ex. `console.warn`) for if an author passes in an incorrect hash?

```markdown
![](/image/with/hash/typo#drk-ony)

☝️ - Similarly would this silently default to just displaying as a regular un-themed image?
```

Aside: content-conformance, might be where we ultimately guard against this. (thinking out loud)

Reply:
yeah that's a good idea, I can add something 👍🏻 And yeah the `data-hide-on-theme` property won't be added if it doesn't match the regex. Have a [little test here](https://github.com/hashicorp/dev-portal/pull/2030/files#diff-a93c7ce6b5c15905f4e1a3132254307b918f422db1a4d833aed4f32eabeed258R21-R27) for that